src/cpp/session/modules/build/SessionInstallRtools.cpp

```diff
@@ -40,36 +41,13 @@ namespace build {
 
 std::string kFallbackUrl = "https://rstudio.org/links/rtools42";
 
-std::string resolveRtools42InstallerUrl(const std::string& url)
-{
-   FilePath tmpFile;
-   FilePath::tempFilePath(tmpFile);
-   Error error = r::exec::RFunction("utils:::download.file")
-         .addParam("url", url)
-         .addParam("destfile", tmpFile.getAbsolutePath())
-         .addParam("mode", "rb")
-         .call();
-   if (error) {
-      LOG_ERROR(error);
-      return kFallbackUrl;
-   }
-
-   std::string rToolsHome = file_utils::readFile(tmpFile);
-   boost::regex reLinkPattern("<a\\shref=\"(.*rtools.*\\.exe)\">\\s*.+<\\/a>", boost::regex::icase);
-   boost::smatch matches;
-
-   if (regex_utils::match(rToolsHome, matches, reLinkPattern))
-      return matches[1];
-   // Use RStudio's hosted copy as a fallback if no match is found
-   return kFallbackUrl;
-}
-
 Error installRtools()
 {
    // populate list of known Rtools installers
    bool gcc49 = module_context::usingMingwGcc49();
    FilePath installPath("C:\\Rtools");
    std::vector<r_util::RToolsInfo> availableRtools;
+   availableRtools.push_back(r_util::RToolsInfo("4.2", installPath, gcc49));
    availableRtools.push_back(r_util::RToolsInfo("4.0", installPath, gcc49));
    availableRtools.push_back(r_util::RToolsInfo("3.5", installPath, gcc49));
    availableRtools.push_back(r_util::RToolsInfo("3.4", installPath, gcc49));
```

Comment:
why are these all gcc49? AFAIK rtools 40 and 42 uses gcc 10, but I don't think this even matters

Reply:
IIRC this is mostly a historical artifact; there was a transition period for older version of Rtools to using gcc 4.9.3 that required a bunch of IDE changes (which IIUC are mostly no longer relevant).